docs/source/util.py

```diff
@@ -0,0 +1,174 @@
+#
+# Copyright (c) 2012-2022 Snowflake Computing Inc. All rights reserved.
+#
+
+import argparse
+import os
+import shutil
+import subprocess
+from collections import namedtuple
+from typing import Iterable, Optional
+
+Class = namedtuple(
+    "Class",
+    ["module", "methods", "attributes"],
+    defaults=["snowflake.snowpark", [], []],
+)
+Module = namedtuple(
+    "Module", ["name", "attributes", "functions", "classes", "exceptions"]
+)
+mapping = {
+    "Methods": "methods",
+    "Attributes": "attributes",
+    "Module Attributes": "attributes",
+    "Functions": "functions",
+    "Classes": "classes",
+    "Exceptions": "exceptions",
+}
+
+
+def autogen_and_parse_for_info(module_name: str, class_name: Optional[str] = None):
+    if class_name:
+        res = Class(module_name, [], [])
+        name = f"{module_name}.{class_name}"
+    else:
+        res = Module(module_name, [], [], [], [])
+        name = module_name
+
+    if not os.path.exists("temp"):
+        os.mkdir("temp")
+
+    rst_content = f"""
+
+.. currentmodule:: snowflake.snowpark
+
+.. autosummary::
+    :members:
+    :inherited-members:
+    :toctree: api/
+
+    {name}
+
+"""
+    fname = f"temp/{name}_temp.rst"
+    with open(fname, "w") as fp:
+        fp.write(rst_content)
+
+    subprocess.run(["sphinx-autogen", fname, "-o", "output", "-t", "_templates"])
+
+    section = ""
+
+    with open(f"output/{name}.rst") as fp:
+        for line in fp:
+            line = line.strip()
+            if line.startswith(".. autosummary::"):
+                continue
+            if line.startswith(".. rubric:: "):
+                line = line[12:]
```

Comment:
Instead of using 12, can we put `len(".. rubric:: ")` instead, or maybe make both `.. rubric:: ` and `.. autosummary:: ` a static constant string variable, so we can do `len(RUBRIC_HEADER)` for example